Core/GDCore/Extensions/PlatformExtension.h

```diff
@@ -225,6 +231,33 @@ class GD_CORE_API PlatformExtension {
                               const gd::String& smallicon_,
                               std::shared_ptr<gd::BaseEvent> instance);
 
+#if defined(GD_IDE_ONLY)
+  /**
+   * \brief Adds a property to the Extension.
+   * \param name The internal name of the property (to get it later).
+   * \param description The description of the property.
+   * \param type The type of the property. Supported ones are "string", "number"
+   * and "boolean".
+   */
+  gd::PropertyDescriptor& RegisterProperty(const gd::String& name,
+                                           const gd::String& description,
+                                           const gd::String& type,
+                                           const gd::String& hint) {
+    extensionPropertiesMetadata[name]
+        .SetDescription(description)
+        .SetType(type)
+        .SetLabel(hint);
+    return extensionPropertiesMetadata[name];
+  };
+#endif
+
+  /**
+   * \brief Adds a property to the Extension.
+   */
+  gd::PropertyDescriptor& RegisterProperty(const gd::String& name) {
+    return extensionPropertiesMetadata[name];
+  };

```

Comment:
To solve the compilation issue, I think you can put this method inside the `#if defined(GD_IDE_ONLY)`/`#endif`.
This effectively removes the existence of this method when the code is compiled for GDCpp (the game engine). It's ok because in the C++ game engine, we don't want to extensions to register things that are "metadata", i.e: that are things related to the editor.

This should solve the compilation.

Reply:
That was it indeed. I forgot I had a "Fluent API" (I think that is how chained call based APIs are called) declaration as well 😅

Reply:
This is indeed the name of this "pattern" where calls to set up something are chained :)